test/cql-pytest/test_allow_filtering.py

```diff
@@ -291,3 +301,63 @@ def test_contains_frozen_collection_ck(cql, test_keyspace):
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS 0 ALLOW FILTERING")))
         assert 1 == len(list(cql.execute(
             "SELECT * FROM " + table + " WHERE a=0 AND c=0 AND b CONTAINS KEY 0 ALLOW FILTERING")))
+
+# table4 contains example table from issue #8991
+@pytest.fixture(scope="module")
+def table4(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (k int, c1 int, c2 int, PRIMARY KEY (k,c1,c2))")
+    cql.execute(f"CREATE INDEX ON {table} (c2)")
+    cql.execute(f"INSERT INTO {table} (k, c1, c2) VALUES (0, 0, 1)")
+    cql.execute(f"INSERT INTO {table} (k, c1, c2) VALUES (0, 1, 1)")
+
+    everything = list(cql.execute(f"SELECT * FROM {table}"))
+    wait_for_index(cql, table, 'c2', everything)
+    yield (table, everything)
+    cql.execute(f"DROP TABLE {table}")
+
+# Selecting from indexed table should using only clustering key should require filtering
+# Variation of #7608 found in #8991
+def test_select_indexed_cluster(cql, table4):
+    check_af_mandatory(cql, table4, 'c1 = 1 AND c2 = 1', lambda row: row.c1 == 1 and row.c2 == 1)
+
+# table5 contains an indexed table with 3 clustering columns.
+# used to test need_filtering() on indexed tables.
+@pytest.fixture(scope="module")
+def table5(cql, test_keyspace):
+    table = test_keyspace + "." + unique_name()
+    cql.execute(f"CREATE TABLE {table} (p int, c1 frozen<list<int>>, c2 frozen<list<int>>, c3 int, PRIMARY KEY (p,c1,c2,c3))")
+    cql.execute(f"CREATE INDEX ON {table} (c3)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, [1], [2], 0)")
+    cql.execute(f"INSERT INTO {table} (p, c1, c2, c3) VALUES (0, [1], [2], 1)")
```

Comment:
This doesn't seem like enough data to test correct filtering.  I'd expect to see multiple rows with the same c3 value, so some can get filtered out.

Reply:
The purpose of these tests is more to test bugs in ALLOW FILTERING needs and less to verify the correctness of filtering, but I agree that if by adding one or a few more rows you can make the correctness test better, it will be even better. Just don't add 100 rows "just in case" - that's just a waste of test time.

Reply:
> The purpose of these tests is more to test bugs in ALLOW FILTERING needs and less to verify the correctness of filtering,

If that's really true, then we've got our priorities wrong.  Correctness should come first (and was the original aim of this PR).

> but I agree that if by adding one or a few more rows you can make the correctness test better, it will be even better.

:+1:

Reply:
I added two more rows that should require filtering in those testcases

Reply:
@dekimir no, nothing is wrong with out priorities: If we find multiple bugs, we should fix all of them, not just the most important one. This specific test on which you're commenting on is for the ALLOW FILTERING bug - not for the correctness bug. Which is perhaps less important, but still a bug.